odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,169 @@ def __repr__(self):
                                          self.__norm.pwisenorm.exponent)
 
 
+class IndicatorSimplex(Functional):
+
+    """The indicator functional of a simplex in any dimension.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \\begin{cases}
+            0 & \\text{if } x_i \geq 0 \\forall i \\text{ and } \sum_i x_i = r
+            \\\\
+            +\\infty & \\text{else.}
+        \\end{cases}
+
+    where `r` is the radius.
+    """
+
+    def __init__(self, space, radius=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        radius : positive float, optional
+            Radius `r` of simplex.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+        self.radius = radius
+
+    def __call__(self, x):
+        """Return whether x is in the unit simplex."""
+
+        r = self.radius
+        if (x.ufuncs.sum() == r and x.ufuncs.greater_equal(0).asarray().all()):
+            return 0
+
+        return np.inf
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional. The indicator functional is not
+        differentiable over the entire domain.
+        """
+
+        raise NotImplementedError('No yet implemented')
+
+    @property
+    def proximal(self):
+        """Return the `proximal factory` of the functional."""
+
+        domain = self.domain
+
+        class ProximalSimplex(Operator):
+            """Proximal operator implemented by the algorithm of [1].
+
+            [1] Duchi, John, et al. "Efficient projections onto the l 1-ball
+            for learning in high dimensions." Proceedings of the 25th
+            international conference on Machine learning. ACM, 2008.
+            """
+
+            def __init__(self, sigma):
+                self.sigma = sigma
+                super().__init__(domain=domain, range=domain, linear=False)
+
+            def _call(self, x, out):
+
+                # projection onto simplex
+                proj_simplex(x, self.radius, out)
+
+        return ProximalSimplex
+
+    @property
+    def convex_conj(self):
+        """The convex conjugate."""
+
+        raise NotImplementedError('Not yet implemented')
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r})'.format(self.__class__.__name__, self.domain)
+
+
+class IndicatorSum(Functional):
```

Comment:
This name is too broad for my taste. It should at least use the word "constraint".